overlord/state/state_test.go

```diff
@@ -546,6 +547,17 @@ func (ss *stateSuite) TestEmptyStateDataAndCheckpointReadAndSet(c *C) {
 
 	// no crash
 	st2.Set("a", 1)
+
+	// ensure all maps of state are correctly initialized by ReadState
+	val := reflect.ValueOf(st2)
+	typ := reflect.TypeOf(val.Elem())
```

Comment:
afaict this needs to be:

	typ := val.Elem().Type()

otherwise typ is reflect.Value and the code after doesn't do what you would think

Reply:
Ah nice, thank you - this is changed